app/src/main/java/com/weatherxm/ui/home/quests/steps/GenericStepActivity.kt

```diff
@@ -0,0 +1,282 @@
+package com.weatherxm.ui.home.quests.steps
+
+import android.Manifest.permission.POST_NOTIFICATIONS
+import android.os.Bundle
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.tooling.preview.Preview
+import com.weatherxm.ui.components.BaseActivity
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.Icon
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import com.weatherxm.ui.components.compose.MediumText
+import com.weatherxm.ui.components.compose.Title
+import com.weatherxm.databinding.ActivityGenericStepBinding
+import androidx.compose.ui.unit.dp
+import com.google.firebase.auth.FirebaseAuth
+import com.weatherxm.R
+import com.weatherxm.ui.Navigator
+import com.weatherxm.ui.common.Contracts.ARG_QUEST_STEP
+import com.weatherxm.ui.common.QuestStep
+import com.weatherxm.ui.common.QuestStepType
+import com.weatherxm.ui.common.ctaButtonTitle
+import com.weatherxm.ui.common.parcelable
+import com.weatherxm.ui.common.stepIcon
+import com.weatherxm.util.hasPermission
+import org.koin.android.ext.android.inject
+import org.koin.androidx.viewmodel.ext.android.viewModel
+import org.koin.core.parameter.parametersOf
+import kotlin.getValue
+
+class GenericStepActivity: BaseActivity() {
+    private val model: GenericStepViewModel by viewModel() {
+        parametersOf(intent.parcelable<QuestStep>(ARG_QUEST_STEP))
+    }
+    private val firebaseAuth: FirebaseAuth by inject()
```

Comment:
No need to add `firebaseAuth` here, let's pass the user ID as an argument with `ARG_USER_ID` I already use and add it via the `parametersOf` above. 😉 

This is more correct in the case also as we have the correct `uid` in the previous screens.